### PR TITLE
docs: add CONTRIBUTING.md file as contributing guide

### PR DESCRIPTION
**Pull request by somratAtSekaiLab**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Contribution guide
+
+### Project tools
+
+- Github for code and issues management and
```
Comment (themitvp) [inline CONTRIBUTING.md:5]: and? 😊 
```suggestion
- Github for code and issues management
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Contribution guide
+
+### Project tools
```
Comment (themitvp) [inline CONTRIBUTING.md:3]: Heading levels should only increment by one level at a time

```suggestion
## Project tools
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+1. Following branches should be used for primary code management
+
+   1. `master` there can be only this branch. This hold all the latest already released code.
+
+   - in case we are doing tag based release, `master` can be used as stable bleeding edge releasable code.
+
+2. `{type}/{GithubIssueNo}-issue-one-liner` should be the format for branch naming
+   1. see [Type](#__type) section for branch `{type}`s.
+   2. Find `{GitgubIssueNo}` in Github.
```
Comment (themitvp) [inline CONTRIBUTING.md:17]:
```suggestion
   2. Find `{GithubIssueNo}` in Github.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+### Branch rules:
+
+1. Following branches should be used for primary code management
+
+   1. `master` there can be only this branch. This hold all the latest already released code.
+
+   - in case we are doing tag based release, `master` can be used as stable bleeding edge releasable code.
+
+2. `{type}/{GithubIssueNo}-issue-one-liner` should be the format for branch naming
+   1. see [Type](#__type) section for branch `{type}`s.
```
Comment (themitvp) [inline CONTRIBUTING.md:16]:
```suggestion
   1. See [Type](#__type) section for branch `{type}`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+   2. Find `{GitgubIssueNo}` in Github.
+
+### Pull requests
+
+Pull requests are the only way to propose a value you want to add. Following is a general workflow for submitting any requests.
+
+1. Clone the repo and create your branch from `master`.
+2. If you've added code that should be tested
+3. Ensure that your code don't failed to build
```
Comment (themitvp) [inline CONTRIBUTING.md:25]:
```suggestion
3. Ensure that your code doesn't fail to build
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+4. Make pull request to `master` branch
+5. Assign respectives to merge pull request
```
Comment (themitvp) [inline CONTRIBUTING.md:27]:
```suggestion
5. Assign respective develops to review and merge pull request
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+### Commit Message Guidelines
+
+We have very precise rules over how our git commit messages can be formatted. This leads to **more
+readable messages** that are easy to follow when looking through the **project history**. But also,
+we use the git commit messages to **generate the change log**.
```
Comment (themitvp) [inline CONTRIBUTING.md:33]:
```suggestion
we use the git commit messages to **generate the changelog**.
```

**Review by themitvp — COMMENTED** [review]
looks good, just a few typos

**Comment by somratAtSekaiLab** [comment]
@themitvp typo fixed :)

**Review by themitvp — APPROVED** [review]
Looks good 🎉

**Pull request merged**